unison-runtime/src/Unison/Runtime/Foreign/Function.hs

```diff
@@ -418,45 +426,55 @@ instance ForeignConvention BufferMode where
   writeForeign stk bm =
     bump stk >>= \stk ->
       case bm of
-        NoBuffering -> stk <$ upoke stk no'buf
-        LineBuffering -> stk <$ upoke stk line'buf
-        BlockBuffering Nothing -> stk <$ upoke stk block'buf
+        NoBuffering -> stk <$ pokeN stk no'buf
+        LineBuffering -> stk <$ pokeN stk line'buf
+        BlockBuffering Nothing -> stk <$ pokeN stk block'buf
         BlockBuffering (Just n) -> do
-          upoke stk n
+          pokeI stk n
           stk <- bump stk
-          stk <$ upoke stk sblock'buf
+          stk <$ pokeN stk sblock'buf
 
 -- In reality this fixes the type to be 'RClosure', but allows us to defer
 -- the typechecker a bit and avoid a bunch of annoying type annotations.
-instance ForeignConvention [Closure] where
+instance {-# OVERLAPPING #-} ForeignConvention [Val] where
```

Comment:
This pragma is just an indication for future readers that there's a more general list instance.